ls1intum/Artemis
# `Programming exercises`: Improve gradle testing template

### Checklist
#### General

### Motivation and Context
This fixes some minor problems with the default gradle testing template.

### Description
1. The SCA-Category `FileTabCharacter` is active by default. This forces students to indent their code with spaces instead of Tabs. Choosing an indentation is mostly considered a choice of personal preference, so this should be left to the student.
2. When ignoring specific issues in specific files, a SuppressionFilter is used. Configuring such a file was problematic since the `confic_loc` variable was never specified. This adds a line to the gradle task to specify this folder and adds an example to the checktyle configuration.
3. While Maven prints output on the standard output streams while testing, Gradle does not do this by default. Using Annotations like `@MirrorOutput` had no effect when used in the context of a Gradle configuration. This now sets `showStandardStreams` to true, which will print such output.

### Steps for Testing
mostly code review 

If you want:

1. Create a Programming Exercise with SCA (Java, Gradle)
2. Check that nothing broke

### Review Progress






#### Code Review
#### Manual Tests
